addons/account/wizard/account_payment_register.py

```diff
@@ -543,6 +543,18 @@ def _get_total_amount_in_wizard_currency_to_full_reconcile(self, batch_result, e
             return abs(residual_amount), False
         else:
             # Foreign currency on payment different than the one set on the journal entries.
+            selected_ids = self.env.context.get('active_ids', [])
+            active_model = self.env.context.get('active_model')
+            if active_model == 'account.move.line':
+                invoices = self.env['account.move.line'].search([('id', 'in', selected_ids)])
+                if invoices:
+                    residual_amount = invoices[0].move_id.amount_residual
+                    return self.source_currency_id._convert(
+                        residual_amount,
+                        self.currency_id,
+                        self.company_id,
+                        self.payment_date,
+                    ), False
```

Comment:
I think you figured it by now but I try to avoid assigning variables that are only used once if they don't grant any clarity so in this case, apart from amount_residual and even that one is arguable, I wouldn't assign any. Personnal preference too, if you resolve the comment, I will check how you changed it but I won't block the PR if you keep it your way.
```suggestion
            if self.env.context.get('active_model') == 'account.move.line':
                if invoices := self.env['account.move.line'].search([('id', 'in', self.env.context.get('active_ids', []))]):
                    residual_amount = invoices[0].move_id.amount_residual
                    return self.source_currency_id._convert(
                        residual_amount,
                        self.currency_id,
                        self.company_id,
                        self.payment_date,
                    ), False
```